src/libstore/store-api.cc

```diff
@@ -864,7 +864,7 @@ StoreType getStoreType(const std::string & uri, const std::string & stateDir)
 {
     if (uri == "daemon") {
         return tDaemon;
-    } else if (uri == "local" || hasPrefix(uri, "/")) {
+    } else if (uri == "local" || hasPrefix(uri, "/") || hasPrefix(uri, "./")) {
```

Comment:
Might make sense to also allow `.` as a path

Reply:
The cost of that would be divergence from the nix expression language though, right?

Reply:
this already happens with cli args. for instance:

```
nix build -f . hello
```

Reply:
but does `-f <url>` work? Because we are overloading URL and relative path syntax, and URL syntax already is infamous for having parser mistakes/ambiguities/vulnerabilities, I rather move slowly. We can always add `.` later.

Of course, if @edolstra wants / there's concensus to go ahead and do `.`, we'll happily add it, but I rather advocate for the smallest change that gets the job done.

Reply:
url's do work (although they have to be in tar format):

```
nix build -f https://github.com/NixOS/nixpkgs/archive/master.tar.gz hello
```

Reply:
rules are:

https://github.com/NixOS/nix/blob/3f01fa1c9c1b5ce0ffd701f8eeb1b16e57aa86d5/src/libstore/filetransfer.cc#L827-L834

Reply:
OK fair, but it as wrong of me to compare `-f <url>` in the first place, because this sort of "each store parses the thing" means having `else` cases is no good either---everything should be `if (...) { ... } else return nullptr`.

Now we could still do `.`, yes. The above does not work argument against that.

Reply:
@edolstra's weak preference is if it doesn't contain `://` and does contain `/` then it is deemed a path and is turned into a LocalStore.

Reply:
In meeting we got @edolstra's preference, and I just updated the PR accordingly.